src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverterBase.cs

```diff
@@ -0,0 +1,327 @@
+
+                scope.Write(field.Id, iConvertible.ToString(CultureInfo.InvariantCulture), @long, @double, ref entryWriter);
+                return;
+
+            case ValueType.Enumerable:
+                var iterator = (IEnumerable)value;
+                if (EnumerableDataStructExist == false)
+                {
+                    StringsListForEnumerableScope = new(InitialSizeOfEnumerableBuffer);
+                    LongsListForEnumerableScope = new(InitialSizeOfEnumerableBuffer);
+                    DoublesListForEnumerableScope = new(InitialSizeOfEnumerableBuffer);
+                    BlittableJsonReaderObjectsListForEnumerableScope = new(InitialSizeOfEnumerableBuffer);
+                }
+
+                var enumerableScope = new EnumerableWriterScope(StringsListForEnumerableScope, LongsListForEnumerableScope, DoublesListForEnumerableScope, BlittableJsonReaderObjectsListForEnumerableScope, _allocator);
+                foreach (var item in iterator)
+                {
+                    InsertRegularField(field, item, indexContext, out _, ref entryWriter, enumerableScope);
+                }
+
+                enumerableScope.Finish(field.Id, ref entryWriter);
+                return;
+
+            case ValueType.DynamicJsonObject:
+                var dynamicJson = (DynamicBlittableJson)value;
+                scope.Write(field.Id, dynamicJson.BlittableJson, ref entryWriter);
+
+                return;
+
+            case ValueType.ConvertToJson:
+                var val = TypeConverter.ToBlittableSupportedType(value);
+                if (val is not DynamicJsonValue json)
+                {
+                    InsertRegularField(field, val, indexContext, out _, ref entryWriter, scope, nestedArray);
+                    return;
+                }
+
+                var jsonScope = Scope.CreateJson(json, indexContext);
+                scope.Write(field.Id, jsonScope, ref entryWriter);
+                return;
+
+            case ValueType.BlittableJsonObject:
+                HandleObject((BlittableJsonReaderObject)value, field, indexContext, out _, ref entryWriter, scope);
+                return;
+
+            case ValueType.DynamicNull:
+                var dynamicNull = (DynamicNullObject)value;
+                if (dynamicNull.IsExplicitNull || _indexImplicitNull)
+                {
+                    scope.Write(field.Id, Encoding.UTF8.GetBytes(Constants.Documents.Indexing.Fields.NullValue), ref entryWriter);
+                }
+                return;
+
+            case ValueType.Null:
+                scope.Write(field.Id, Encoding.UTF8.GetBytes(Constants.Documents.Indexing.Fields.NullValue), ref entryWriter);
+                return;
+            case ValueType.BoostedValue:
+                throw new InvalidParameterException("Boosting in index is not supported by Corax. You can do it during querying or change index type into Lucene.");
```

Comment:
1. `NotSupportedException` would be more suitable here.
2. Please add TODO and reference: https://issues.hibernatingrhinos.com/issue/RavenDB-18146